terraform/deployments/rds/create-encrypted-snapshots.sh

```diff
@@ -0,0 +1,297 @@
+    if ! create_unencrypted_snapshot "$DB" "$UNENCRYPTED_SNAPSHOT_NAME"; then
+      >&2 echo "$DB: FAILED!"
+      echo -e "$(colour_red "$DB: creating unencrytped snapshot FAILED!")"
+      return 1
+    fi
+  fi
+
+  echo "$DB: waiting for snapshot $UNENCRYPTED_SNAPSHOT_NAME to complete"
+  local STATE
+  if ! STATE=$(wait_for_snapshot_terminal_state "$DB" "$UNENCRYPTED_SNAPSHOT_NAME"); then
+    >&2 echo "$DB: FAILED to query state of snapshot"
+    echo -e "$(colour_red "$DB: FAILED to query state of snapshot")"
+    return 1
+  fi
+
+  if [ "$STATE" == "available" ]; then
+    echo -e "$(colour_green "$DB: successfully created snapshot")"
+  elif [ "$STATE" == "failed" ]; then
+    >&2 echo "$DB: snapshot creation FAILED!"
+    echo -e "$(colour_red "$DB: snapshot creation FAILED!")"
+    return 1
+  else
+    >&2 echo "$DB: Unknown final state of snapshot ${STATE}!"
+    echo -e "$(colour_red "$DB: Unknown final state of snapshot ${STATE}!")"
+    return 1
+  fi
+
+  if snapshot_already_exists "$DB" "$ENCRYPTED_SNAPSHOT_NAME"; then
+    echo "$DB: snapshot copy $ENCRYPTED_SNAPSHOT_NAME already exists..."
+  else
+    echo "$DB: snapshot copy doesn't exist, creating"
+    if ! copy_to_encrypted_snapshot "$DB" "$UNENCRYPTED_SNAPSHOT_NAME" "$ENCRYPTED_SNAPSHOT_NAME"; then
+      >&2 echo "$DB: FAILED  to create encrypted snapshot copy!"
+      echo -e "$(colour_red "$DB: FAILED  to create encrypted snapshot copy!")"
+      return 1
+    fi
+  fi          
+
+  echo "$DB: waiting for snapshot copy from $UNENCRYPTED_SNAPSHOT_NAME to $ENCRYPTED_SNAPSHOT_NAME to complete"
+  if ! STATE=$(wait_for_snapshot_terminal_state "$DB" "$ENCRYPTED_SNAPSHOT_NAME"); then
+    >&2 echo "$DB: FAILED to query state of snapshot"
+    echo -e "$(colour_red "$DB: FAILED to query state of snapshot")"
+    return 1
+  fi
+
+  if [ "$STATE" == "available" ]; then
+    echo -e "$(colour_green "$DB: successfully created snapshot copy. COMPLETE")"
+  elif [ "$STATE" == "failed" ]; then
+    >&2 echo "$DB: snapshot copy creation FAILED!"
+    echo -e "$(colour_red "$DB: snapshot copy creation FAILED!")"
+    return 1
+  else
+    >&2 echo "$DB: Unknown final state of snapshot copy ${STATE}!"
+    echo -e "$(colour_red "$DB: Unknown final state of snapshot copy ${STATE}!")"
+    return 1
+  fi
+}
+export -f perform_snapshot
+
+confirm_snapshots
+
+FAILURE_LOG=$(mktemp -t create-snapshot-error-log)
+FULL_LOG_FILE=$(mktemp -t create-snapshot-log)
+echo "STDERR log file, tail this if you want to see errors and also info output showing the current snapshot status while waiting for them to complete: $FAILURE_LOG"
+
+if ! parallel --line-buffer perform_snapshot {} ::: "${DBS[@]}" 2>> "$FAILURE_LOG" | tee -a "$FULL_LOG_FILE"; then
+  echo "-----------------------------------------------------------------------------------------------"
+  echo "ERRORS OCCURED: Some errors occured, see failure log file: $FAILURE_LOG"
```

Comment:
[nitpick] "ERRORS OCCURED" should be "ERRORS OCCURRED"

Reply:
Doh! In vim I have spellcheck running, apparently not in helix :)